src/block.c

```diff
@@ -494,7 +494,8 @@ coap_add_data_large_internal(coap_session_t *session,
       coap_update_token(pdu, lg_xmit->b.b1.token_length, lg_xmit->b.b1.token);
       coap_update_option(pdu,
                          COAP_OPTION_SIZE1,
-                         coap_encode_var_safe(buf, sizeof(buf), length),
+                         coap_encode_var_safe(buf, sizeof(buf),
+                                              (uint32_t)length),
```

Comment:
`coap_encode_var_safe()` takes an `unsigned int` as third parameter (_value_). Therefore, these casts should be to `unsigned int` rather than `uint32_t`.

Reply:
Updated and code pushed.